example/mqttapp/mqtt-example.c

```diff
@@ -183,10 +184,15 @@ int mqtt_client_example(void)
 
     mqtt.event_handler = smartled_event_handler;
     mqtt.delete_subdev = NULL;
-    if (mqtt_init_instance(mqtt.productKey, mqtt.deviceName, mqtt.deviceSecret, mqtt.max_msg_size) < 0) {
-        LOG("mqtt_init_instance failed\n");
-        return -1;
-    }
+    while (retry--)
```

Comment:
If this is indeed a problem, we'd better create a standalone pull request to deal with it. BTW please remove personal debug print.

Reply:
Sometimes, the MQTT instance cannot be created due to connack timeout from MQTT broker.  It is a workaround to avoid the issue. 

LOG:
[hw_start_hal]
100 full
TCP/IP initialized.
trace config close!!!
[001720]<V> aos framework init.
post got ip event ,ip is 0x100ba8c0
[004504]<V> wifi_service_event!
[inf] iotx_device_info_init(40): device_info created successfully!
[dbg] iotx_device_info_set(50): start to set device info!
[dbg] iotx_device_info_set(64): device_info set successfully!
[dbg] guider_print_dev_guider_info(248): ....................................................
[dbg] guider_print_dev_guider_info(249):           ProductKey : a1bJzuy23YE
[dbg] guider_print_dev_guider_info(250):           DeviceName : test_M487JIDAE_dev_01
[dbg] guider_print_dev_guider_info(251):             DeviceID : a1bJzuy23YE.test_M487JIDAE_dev_01
[dbg] guider_print_dev_guider_info(253): ....................................................
[dbg] guider_print_dev_guider_info(254):        PartnerID Buf : ,partner_id=AliOSThings
[dbg] guider_print_dev_guider_info(255):         ModuleID Buf : ,module_id=GeneralID
[dbg] guider_print_dev_guider_info(256):           Guider URL :
[dbg] guider_print_dev_guider_info(258):       Guider SecMode : 3 (TCP + Direct + Plain)
[dbg] guider_print_dev_guider_info(260):     Guider Timestamp : 2524608000000
[dbg] guider_print_dev_guider_info(261): ....................................................
[dbg] guider_print_dev_guider_info(267): ....................................................
[dbg] guider_print_conn_info(225): -----------------------------------------
[dbg] guider_print_conn_info(226):             Host : a1bJzuy23YE.iot-as-mqtt.cn-shanghai.aliyuncs.com
[dbg] guider_print_conn_info(227):             Port : 1883
[dbg] guider_print_conn_info(230):         ClientID : a1bJzuy23YE.test_M487JIDAE_dev_01|securemode=3,timestamp=2524608000000,signmethod=hmacsha1,gw=0,partner_id=AliOSThings,module_id=GeneralID|
[dbg] guider_print_conn_info(235): -----------------------------------------
[inf] iotx_mc_init(1709): MQTT init success!
RHINOSOCK 32 HAL_TCP_Establish() | establish tcp connection with server(host=a1bJzuy23YE.iot-as-mqtt.cn-shanghai.aliyuncs.com port=1883)
RHINOSOCK 71 HAL_TCP_Establish() | success to establish tcp, fd=0
[dbg] connect_tcp(75): sockfd: 0
[err] iotx_mc_wait_CONNACK(1209): wait connack timeout
[err] iotx_mc_connect(2063): wait connect ACK timeout, or receive a ACK indicating error!
[inf] iotx_mc_disconnect(2178): mqtt disconnect!
[inf] iotx_mc_release(2224): mqtt release!
[009102]<V> mqtt_init_instance failed